src/dirext.rs

```diff
@@ -713,16 +737,20 @@ impl CapStdExtDirExt for Dir {
         let mut t = cap_tempfile::TempFile::new(&d)?;
         // Apply the permissions, if we have them
         if let Some(existing_perms) = existing_perms {
-            t.as_file_mut().set_permissions(existing_perms)?;
+            t.as_file().set_permissions(existing_perms)?;
```

Comment:
Which, Hmm OK we were not quite doing this right before actually in theory; we should apply a restrictive umask and then set the permissions after the write is done.

Anyways backing aup a second, why are you making this change? It's OK, I just want to understand why.

Reply:
yeah but the closure might have changed the permissions, so we can't reapply the permissions after :(

I'll move the permissions changes in a separate PR as suggested